### ISSUE TITLE
Issue: viaNot is now reporting wrong cycles

### ISSUE THREAD

**Issue by epaillous** [sverweij/dependency-cruiser#604]
### Expected Behavior

I updated dependency cruiser to last version 11.7.1, I have now new failures on `viaNot`, because it reports cycle when testing "knots" that I wanted to exclude. 
Here is my rule : 

```
  {
    name: 'no-circular',
    severity: 'error',
    comment: `This dependency is part of a circular relationship.
      You might want to revise your solution (i.e. use dependency inversion,
       make sure the modules have a single responsibility) `,
    from: {},
    to: {
      circular: true,
      viaNot: [
        '^app/javascript/tables/index.ts',
        '^app/javascript/ui/index.tsx',
        '^app/javascript/hooks/index.tsx',
        '^app/javascript/deprecated_types/index.ts',
        'spec/javascript/factory/index.ts',
      ],
    },
  },
```

Before 11.7.1, I did not have any errors. I expect "knot files" to be ignored by the rule since they appear in the `viaNot`.

### Current Behavior

I have now some errors like : 
```
  error no-circular: app/javascript/ui/index.tsx → 
      app/javascript/ui/Checkbox.tsx →
      app/javascript/ui/index.tsx
  error no-circular: app/javascript/ui/index.tsx → 
      app/javascript/ui/Breadcrumb.tsx →
      app/javascript/patterns/PageHeader.tsx →
      app/javascript/ui/index.tsx
```

So basically, it is now reporting errors when testing circular inside "knots file".

### Possible Solution

As a workaround, I added all these files to `pathNot` in `from` : 
```
    {
      name: 'no-circular',
      severity: 'error',
      comment: `This dependency is part of a circular relationship.
        You might want to revise your solution (i.e. use dependency inversion,
         make sure the modules have a single responsibility) `,
      from: {
        pathNot: [
          '^app/javascript/tables/index.ts',
          '^app/javascript/ui/index.tsx',
          '^app/javascript/hooks/index.tsx',
          '^app/javascript/deprecated_types/index.ts',
          'spec/javascript/factory/index.ts',
        ],
      },
      to: {
        circular: true,
        viaNot: [
          '^app/javascript/tables/index.ts',
          '^app/javascript/ui/index.tsx',
          '^app/javascript/hooks/index.tsx',
          '^app/javascript/deprecated_types/index.ts',
          'spec/javascript/factory/index.ts',
        ],
      },
    },
```

Is it an expected behaviour or a bug ? 

### Your Environment

- Version used: 11.7.1
- Node version: 16.3.0
- Operating System and version: MacOS Monterey 12.0.1

**Comment by sverweij** [comment]
Hi @epaillous this looks like a bug, thanks for raising it!

 Recently some code changed around _via_ like rules - and it looks like one or more non-regression tests are missing. I'll make a minimal reproduction sample & fix/ revert where necessary.

**Comment by sverweij** [comment]
Hi - found the bug and fixed it (see linked PR #605 above). It was introduced in 11.7.0 (with PR #589/ commit b71566d1c). 

Will release the fix before the end of the weekend in v11.8.0.
